components/ReportList.vue

```diff
@@ -108,7 +124,7 @@ const getClasses = (dep) => {
     <business-button
       :disabled="isLoading"
       class="inline-flex items-center rounded-none text-slate-50 fixed md:sticky bottom-0 w-full justify-center bg-green-600 hover:bg-green-700 z-10"
-      @click="commit"
+      @click="modalOpen = true"
```

Comment:
**suggestion:** Let's put this in a named function. You could do a single called `toggleModal` and then refer to it in each spot we are toggling the state (in `onSubmit` and `onClose`).

Reply:
done

Reply:
I think I did not describe this clearly, so I added a commit that has everything referring to a single `toggleModal` function.